Sources/BrowserServicesKit/Common/Extensions/URLExtension.swift

```diff
@@ -74,12 +74,17 @@ extension URL {
 
     public func addParameter(name: String, value: String) throws -> URL {
         guard var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else { throw ParameterError.parsingFailed }
-        var queryItems = components.queryItems ?? [URLQueryItem]()
+        var encodedQueryItems = components.percentEncodedQueryItems ?? [URLQueryItem]()
+
         let newQueryItem = URLQueryItem(name: name, value: value)
-        queryItems.append(newQueryItem)
-        components.queryItems = queryItems
+        components.queryItems = [newQueryItem]
```

Comment:
Instead of doing this, we could probably just call `addingPercentEncoding()` on the `name` and `value`. This would make the implementation simpler, but in my approach we rely on `URLComponents` to perform percent encoding. I belive it's a bit safer approach (more future-proof since we expect `URLComponents` to do the encoding properly), but please let me know if I'm being too cautious.